$NetBSD: patch-Makefile,v 1.2 2025/10/03 18:14:19 dkazankov Exp $

Provide gprbuild options
Run testsuite in WRKDIR instead of /tmp

--- Makefile.orig	2024-10-07 18:46:25.000000000 +0300
+++ Makefile
@@ -1,12 +1,14 @@
 BUILD_MODE ?= dev
 LIBRARY_TYPE ?= relocatable
 PROCESSORS ?= 0
+GPRBUILD_OPTIONS ?= -p -j$(PROCESSORS)
 
 ALL_LIBRARY_TYPES = static static-pic relocatable
 
 LIB_PROJECT = prettier_ada.gpr
 
 TEST_PROGRAMS = testsuite/test_programs/test_programs.gpr
+TESTSUITE_DIR := $(shell pwd)/testsuite
 
 .PHONY: lib
 lib:
@@ -17,8 +19,7 @@
 		-XPRETTIER_ADA_LIBRARY_TYPE=$(LIBRARY_TYPE) \
 		-XPRETTIER_ADA_BUILD_MODE=$(BUILD_MODE) \
 		-P $(LIB_PROJECT) \
-		-p \
-		-j$(PROCESSORS) ; \
+		$(GPRBUILD_OPTIONS) ;
 
 .PHONY: all
 all:
@@ -30,8 +31,7 @@
 			-XLIBRARY_TYPE=$$library_type \
 			-XPRETTIER_ADA_BUILD_MODE=$(BUILD_MODE) \
 			-P $(LIB_PROJECT) \
-			-p \
-			-j$(PROCESSORS) ; \
+			$(GPRBUILD_OPTIONS) ; \
 	done;
 
 .PHONY: clean
@@ -96,4 +96,6 @@
 
 .PHONY: test
 test: test-programs
-	python3 testsuite/testsuite.py
+	TEST_PROGRAMS=$(TESTSUITE_DIR)/test_programs/bin python3 testsuite/testsuite.py \
+	    -t $(TESTSUITE_DIR)/temp -d $(TESTSUITE_DIR)/dest -o $(TESTSUITE_DIR)/out \
+	    --no-random-temp-subdir --cleanup-mode all --show-error-output
